src/org/labkey/test/components/react/dialogs/BulkCreateSamplesDialog.java

```diff
@@ -0,0 +1,160 @@
+package org.labkey.test.components.react.dialogs;
+
+import org.labkey.test.BootstrapLocators;
+import org.labkey.test.Locator;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.labkey.test.components.glassLibrary.components.FilteringReactSelect;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_JAVASCRIPT;
+
+public class BulkCreateSamplesDialog extends ModalDialog
+{
+
+    public BulkCreateSamplesDialog(WebDriver driver)
+    {
+        this(new ModalDialogFinder(driver).withTitle("Bulk Creation of Samples"));
+    }
+
+    private BulkCreateSamplesDialog(ModalDialogFinder finder)
+    {
+        super(finder);
+    }
+
+    public BulkCreateSamplesDialog setQuantity(int quantity)
+    {
+        return setQuantity(Integer.toString(quantity));
+    }
+
+    public BulkCreateSamplesDialog setQuantity(String quantity)
+    {
+        getWrapper().setFormElement(elementCache().quantity, quantity);
+        return this;
+    }
+
+    public String getQuantity()
+    {
+        return getWrapper().getFormElement(elementCache().quantity);
+    }
+
+    public BulkCreateSamplesDialog setDescription(String description)
+    {
+        getWrapper().setFormElement(elementCache().description, description);
+        return this;
+    }
+
+    public String getDescription()
+    {
+        return getWrapper().getFormElement(elementCache().description);
+    }
+
+    public BulkCreateSamplesDialog setTextField(String fieldCaption, String value)
+    {
+        String forField = Locator.xpath("//div[@class='modal-body']//label[./span[contains(text(), '" + fieldCaption + "')]]").findElement(getComponentElement()).getAttribute("for");
+        getWrapper().setFormElement(Locator.tagWithId("input", forField), value);
+        return this;
+    }
+
+    public String getTextField(String fieldCaption)
+    {
+        String forField = Locator.xpath("//div[@class='modal-body']//label[./span[contains(text(), '" + fieldCaption + "')]]").findElement(getComponentElement()).getAttribute("for");
+        return getWrapper().getFormElement(Locator.tagWithId("input", forField));
+    }
+
+    private boolean waitForReactSelectReady(FilteringReactSelect select)
+    {
+        getWrapper().waitFor(()->!select.getSelections().contains("Loading..."), "The reactSelect field did not populate in time.", 1_000);
+        return true;
+    }
```

Comment:
This component shouldn't need to determine whether the select is ready and it certainly shouldn't need to know how to do so.

Reply:
I agree. This was a hack/work-around for an issue with the ReactSelect. I'll start an email thread to discuss best solutions.